### ISSUE TITLE
A error about slangtorch

### ISSUE THREAD

**Issue by hsjsbxkkamabsh** [chinmay0301ucsd/Vol3DGS#10]
My install process is sucessful but there is a error when running train.py. It seems a error about slangtorch when loading .slang document.

<img width="1315" height="341" alt="Image" src="https://github.com/user-attachments/assets/985ab459-453a-4d2a-a041-6840be4df68b" />

**Comment by chinmay0301ucsd** [comment]
Can you provide more details on the conda environment / the path variables mentioned in the README?

**Comment by hsjsbxkkamabsh** [comment]
My packege in conda emvironment: 
Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
absl-py                   2.3.1                    pypi_0    pypi
addict                    2.4.0                    pypi_0    pypi
annotated-types           0.7.0                    pypi_0    pypi
asttokens                 3.0.0                    pypi_0    pypi
attrs                     25.3.0                   pypi_0    pypi
blinker                   1.9.0                    pypi_0    pypi
bzip2                     1.0.8                hda65f42_8    conda-forge
ca-certificates           2025.8.3             hbd8a1cb_0    conda-forge
certifi                   2025.8.3                 pypi_0    pypi
charset-normalizer        3.4.3                    pypi_0    pypi
click                     8.1.8                    pypi_0    pypi
cmake                     4.1.0                    pypi_0    pypi
comm                      0.2.3                    pypi_0    pypi
configargparse            1.7.1                    pypi_0    pypi
contourpy                 1.3.0                    pypi_0    pypi
cuda-cccl                 12.4.127                      0    nvidia
cuda-cudart               12.4.127                      0    nvidia
cuda-cudart-dev           11.7.99                       0    nvidia
cuda-nvcc                 11.7.99                       0    nvidia
cudatoolkit               11.7.1              h4bc3d14_13    conda-forge
cycler                    0.12.1                   pypi_0    pypi
cython                    0.29.36                  pypi_0    pypi
dash                      3.2.0                    pypi_0    pypi
decorator                 5.2.1                    pypi_0    pypi
diff-gaussian-rasterization 0.0.0                    pypi_0    pypi
eval-type-backport        0.2.2                    pypi_0    pypi
exceptiongroup            1.3.0                    pypi_0    pypi
executing                 2.2.1                    pypi_0    pypi
fastjsonschema            2.21.2                   pypi_0    pypi
filelock                  3.19.1                   pypi_0    pypi
flask                     3.1.2                    pypi_0    pypi
fonttools                 4.59.2                   pypi_0    pypi
fused-ssim                0.0.0                    pypi_0    pypi
gitdb                     4.0.12                   pypi_0    pypi
gitpython                 3.1.45                   pypi_0    pypi
grpcio                    1.74.0                   pypi_0    pypi
hatchling                 1.27.0                   pypi_0    pypi
idna                      3.10                     pypi_0    pypi
importlib-metadata        8.7.0                    pypi_0    pypi
importlib-resources       6.5.2                    pypi_0    pypi
intel-cmplr-lib-ur        2024.2.1                 pypi_0    pypi
intel-openmp              2024.2.1                 pypi_0    pypi
ipython                   8.18.1                   pypi_0    pypi
ipywidgets                8.1.7                    pypi_0    pypi
itsdangerous              2.2.0                    pypi_0    pypi
jedi                      0.19.2                   pypi_0    pypi
jinja2                    3.1.6                    pypi_0    pypi
joblib                    1.5.2                    pypi_0    pypi
jsonschema                4.25.1                   pypi_0    pypi
jsonschema-specifications 2025.9.1                 pypi_0    pypi
jupyter-core              5.8.1                    pypi_0    pypi
jupyterlab-widgets        3.0.15                   pypi_0    pypi
kiwisolver                1.4.7                    pypi_0    pypi
ld_impl_linux-64          2.44                 h1423503_1    conda-forge
libblas                   3.9.0           35_h4a7cf45_openblas    conda-forge
libcblas                  3.9.0           35_h0358290_openblas    conda-forge
libffi                    3.4.6                h2dba641_1    conda-forge
libgcc                    15.1.0               h767d61c_5    conda-forge
libgcc-ng                 15.1.0               h69a702a_5    conda-forge
libgfortran               15.1.0               h69a702a_5    conda-forge
libgfortran5              15.1.0               hcea5267_5    conda-forge
libgomp                   15.1.0               h767d61c_5    conda-forge
liblapack                 3.9.0           35_h47877c9_openblas    conda-forge
liblzma                   5.8.1                hb9d3cd8_2    conda-forge
liblzma-devel             5.8.1                hb9d3cd8_2    conda-forge
libnsl                    2.0.1                hb9d3cd8_1    conda-forge
libopenblas               0.3.30          pthreads_h94d23a6_2    conda-forge
libsqlite                 3.50.4               h0c1763c_0    conda-forge
libstdcxx                 15.1.0               h8f9b012_5    conda-forge
libstdcxx-ng              15.1.0               h4852527_5    conda-forge
libuuid                   2.41.1               he9a06e4_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libzlib                   1.3.1                hb9d3cd8_2    conda-forge
lit                       18.1.8                   pypi_0    pypi
markdown                  3.9                      pypi_0    pypi
markupsafe                3.0.2                    pypi_0    pypi
matplotlib                3.9.4                    pypi_0    pypi
matplotlib-inline         0.1.7                    pypi_0    pypi
mkl                       2024.0.0                 pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
narwhals                  2.4.0                    pypi_0    pypi
nbformat                  5.10.4                   pypi_0    pypi
ncurses                   6.5                  h2d0b736_3    conda-forge
nest-asyncio              1.6.0                    pypi_0    pypi
networkx                  3.2.1                    pypi_0    pypi
ninja                     1.13.0                   pypi_0    pypi
numpy                     1.26.4                   pypi_0    pypi
nvidia-cublas-cu11        11.10.3.66               pypi_0    pypi
nvidia-cuda-cupti-cu11    11.7.101                 pypi_0    pypi
nvidia-cuda-nvrtc-cu11    11.7.99                  pypi_0    pypi
nvidia-cuda-runtime-cu11  11.7.99                  pypi_0    pypi
nvidia-cudnn-cu11         8.5.0.96                 pypi_0    pypi
nvidia-cufft-cu11         10.9.0.58                pypi_0    pypi
nvidia-curand-cu11        10.2.10.91               pypi_0    pypi
nvidia-cusolver-cu11      11.4.0.1                 pypi_0    pypi
nvidia-cusparse-cu11      11.7.4.91                pypi_0    pypi
nvidia-nccl-cu11          2.14.3                   pypi_0    pypi
nvidia-nvtx-cu11          11.7.91                  pypi_0    pypi
open3d                    0.18.0                   pypi_0    pypi
opencv-python             4.11.0.86                pypi_0    pypi
openssl                   3.5.2                h26f9b46_0    conda-forge
packaging                 25.0                     pypi_0    pypi
pandas                    2.3.2                    pypi_0    pypi
parso                     0.8.5                    pypi_0    pypi
pathspec                  0.12.1                   pypi_0    pypi
pexpect                   4.9.0                    pypi_0    pypi
pillow                    11.3.0                   pypi_0    pypi
pip                       24.2               pyh8b19718_1    conda-forge
platformdirs              4.4.0                    pypi_0    pypi
plotly                    6.3.0                    pypi_0    pypi
pluggy                    1.6.0                    pypi_0    pypi
plyfile                   1.1                pyhd8ed1ab_0    conda-forge
pooch                     1.8.2                    pypi_0    pypi
prompt-toolkit            3.0.52                   pypi_0    pypi
protobuf                  6.32.0                   pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pure-eval                 0.2.3                    pypi_0    pypi
pydantic                  2.11.7                   pypi_0    pypi
pydantic-core             2.33.2                   pypi_0    pypi
pydicom                   2.4.4                    pypi_0    pypi
pygments                  2.19.2                   pypi_0    pypi
pyparsing                 3.2.3                    pypi_0    pypi
pyquaternion              0.9.9                    pypi_0    pypi
python                    3.9.20          h13acc7a_1_cpython    conda-forge
python-dateutil           2.9.0.post0              pypi_0    pypi
python_abi                3.9                      8_cp39    conda-forge
pytz                      2025.2                   pypi_0    pypi
pyvista                   0.46.3                   pypi_0    pypi
pyyaml                    6.0.2                    pypi_0    pypi
readline                  8.2                  h8c095d6_2    conda-forge
referencing               0.36.2                   pypi_0    pypi
requests                  2.32.5                   pypi_0    pypi
retrying                  1.4.2                    pypi_0    pypi
rpds-py                   0.27.1                   pypi_0    pypi
scikit-learn              1.6.1                    pypi_0    pypi
scipy                     1.13.1                   pypi_0    pypi
scooby                    0.10.1                   pypi_0    pypi
sentry-sdk                2.37.1                   pypi_0    pypi
setuptools                80.9.0             pyhff2d567_0    conda-forge
simple-knn                0.0.0                    pypi_0    pypi
six                       1.17.0                   pypi_0    pypi
slang-gaussian-rasterization 0.1.0                    pypi_0    pypi
slangtorch                1.3.12                   pypi_0    pypi
smmap                     5.0.2                    pypi_0    pypi
stack-data                0.6.3                    pypi_0    pypi
sympy                     1.14.0                   pypi_0    pypi
tbb                       2021.13.1                pypi_0    pypi
tensorboard               2.20.0                   pypi_0    pypi
tensorboard-data-server   0.7.2                    pypi_0    pypi
tensorboardx              2.6.4                    pypi_0    pypi
threadpoolctl             3.6.0                    pypi_0    pypi
tifffile                  2024.8.30                pypi_0    pypi
tk                        8.6.13          noxft_hd72426e_102    conda-forge
tomli                     2.2.1                    pypi_0    pypi
torch                     2.0.1                    pypi_0    pypi
torchaudio                2.0.2                    pypi_0    pypi
torchvision               0.15.2                   pypi_0    pypi
tqdm                      4.67.1                   pypi_0    pypi
traitlets                 5.14.3                   pypi_0    pypi
triton                    2.0.0                    pypi_0    pypi
trove-classifiers         2025.9.9.12              pypi_0    pypi
typing-extensions         4.15.0                   pypi_0    pypi
typing-inspection         0.4.1                    pypi_0    pypi
tzdata                    2025.2                   pypi_0    pypi
urllib3                   2.5.0                    pypi_0    pypi
vtk                       9.5.1                    pypi_0    pypi
wandb                     0.21.3                   pypi_0    pypi
wcwidth                   0.2.13                   pypi_0    pypi
werkzeug                  3.1.3                    pypi_0    pypi
wheel                     0.45.1             pyhd8ed1ab_1    conda-forge
widgetsnbextension        4.0.14                   pypi_0    pypi
xz                        5.8.1                hbcc6ac9_2    conda-forge
xz-gpl-tools              5.8.1                hbcc6ac9_2    conda-forge
xz-tools                  5.8.1                hb9d3cd8_2    conda-forge
zipp                      3.23.0                   pypi_0    pypi

................................................................................................................................
my cudatoolkit version:
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Jun__8_16:49:14_PDT_2022
Cuda compilation tools, release 11.7, V11.7.99
Build cuda_11.7.r11.7/compiler.31442593_0
.................................................................................................................................
my python version is 3.9.20; my Ubuntu version is 20.04; my GPU is 3090 with driver version 535.104.05

**Comment by chinmay0301ucsd** [comment]
Can you try with slangtorch 1.3.7? The env says 1.3.12 . Also did you check if the slangtorch toy examples work for you? https://shader-slang.org/slang/docs/user-guide/a1-02-slangpy

**Comment by hsjsbxkkamabsh** [comment]
It's no work. It outputs the error at "m=slangtorch.loadModule(...)". Should I need to install slang in my compuer? I haven't installed it yet. 
<img width="523" height="32" alt="Image" src="https://github.com/user-attachments/assets/1200ef2c-8326-4c9b-8d06-34a644d13ab4" />

**Comment by chinmay0301ucsd** [comment]
slangtorch installation should install slang as well iirc.

**Comment by chinmay0301ucsd** [comment]
Also did you use the same instructions as provided in the readme?

**Comment by hsjsbxkkamabsh** [comment]
Yes, I used the same instruction as readme. Thank you, I got it. Maybe I need to install slang.
